machina/apps/forum_search/search_indexes.py

```diff
@@ -35,7 +36,10 @@ def get_model(self):
         return Post
 
     def prepare_poster_name(self, obj):
-        return obj.poster.username if obj.poster else obj.username
+        return (obj.poster.username + ((' ' + obj.poster.first_name + ' ' + obj.poster.last_name)
```

Comment:
Why make the assumption that the `first_name`/ `last_name` fields will always be available on the user instance here? If a custom `User` model is used without these fields defined this code wouldn't work at all.

Reply:
We added templates because the user object doesn't really have that much customisability in it. With a template you can add anything you like as long as you make sure it's in the context.

As for the search, this one depends on a separate setting. So if somebody uses a custom user model, they might want to avoid setting this to True if they don't have the fields.